### PR TITLE
Unify block_dev

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/drivers/block_dev/block_dev.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+extern struct idesc_ops idesc_bdev_ops;
+extern int devfs_add_block(struct block_dev *dev);
+extern int devfs_del_block(struct block_dev *dev);
+
+struct block_dev *block_dev_create(const char *path, const struct block_dev_driver *driver, void *privdata) {
+	struct block_dev *bdev;
+	struct dev_module *devmod;
+
+	bdev = block_dev_create_common((char *)basename((char *)path), driver, privdata);
```
Comment (Deryugin) [inline src/drivers/block_dev/block_dev.c]: Can we now merge `block_dev_create()` and `block_dev_create_common()` into one function?

Comment (anton-bondarev) [reply src/drivers/block_dev/block_dev.c]: Yes, I think so

File: src/fs/driver/devfs/devfs_dvfs.c [hunk 2]
```diff
@@ -287,3 +288,25 @@ static struct auto_mount devfs_auto_mount = {
 
 ARRAY_SPREAD_DECLARE(const struct auto_mount *const, auto_mount_tab);
 ARRAY_SPREAD_ADD(auto_mount_tab, &devfs_auto_mount);
+
+int devfs_add_block(struct block_dev *bdev) {
+	char full_path[DVFS_MAX_PATH_LEN];
```
Comment (Deryugin) [inline src/fs/driver/devfs/devfs_dvfs.c]: Seems like this function doesnt' create a new node in VFS tree.

File: src/fs/driver/devfs/devfs_oldfs.c [hunk 3]
```diff
@@ -59,3 +60,40 @@ static const struct fs_driver devfs_driver = {
...
+
+int devfs_add_block(struct block_dev *bdev) {
+	struct path node, root;
+	struct nas *nas;
+	struct node_fi *node_fi;
+	char full_path[PATH_MAX];
+
+	vfs_get_root_path(&root);
+
+	/* XXX  Workaround for compatibility with DVFS.
```
Comment (Deryugin) [inline src/fs/driver/devfs/devfs_oldfs.c]: I think that this comment is irrelevant now and can be deleted.

File: src/fs/dvfs/dvfs_str.c [hunk 4]
```diff
@@ -26,32 +26,36 @@ const char *dvfs_last_link(const char *path) {
...
-	while (*l != '/' && l > path)
+	while (*l != '/' && l > path) {
 		l--;
+	}
 
 	return l;
 }
 
-
+#if 0
```
Comment (Deryugin) [inline src/fs/dvfs/dvfs_str.c]: I think that this function is trivial enough to be safely removed.

File: src/fs/syslib/kfile.c [hunk 5]
```diff
@@ -57,7 +57,7 @@ struct idesc *kopen(struct node *node, int flag) {
 			return NULL;
 		}
 
-		if (S_ISCHR(node->mode)) {
+		if (S_ISCHR(node->mode) || S_ISBLK(node->mode)) {
 			return char_dev_open(node, flag);
```
Comment (Deryugin) [inline src/fs/syslib/kfile.c]: Calling `char_dev_open()` for block device may look like a typo. I think that it's a good idea to write comment here or rename function itself.

File: src/fs/driver/devfs/devfs_oldfs.c (hunk 3 above)
Comment (anton-bondarev) [reply src/fs/driver/devfs/devfs_oldfs.c]: done

File: src/fs/dvfs/dvfs_str.c (hunk 4 above)
Comment (anton-bondarev) [reply src/fs/dvfs/dvfs_str.c]: done.
Rm the whole file

File: src/fs/syslib/kfile.c (hunk 5 above)
Comment (anton-bondarev) [reply src/fs/syslib/kfile.c]: done.
Add a comment bellow

File: src/fs/driver/devfs/devfs_dvfs.c (hunk 2 above)
Comment (anton-bondarev) [reply src/fs/driver/devfs/devfs_dvfs.c]: done

File: src/compat/libc/stdio/rename_dvfs.c [hunk 6]
```diff
@@ -56,9 +57,10 @@ int rename(const char *src_name, const char *dst_name) {
 		return SET_ERRNO(EISDIR);
 
 	/* Some ugly string handling, it's better to rewrite it somehow */
+	/* May be it's better to use dirname() instead of */
```
Comment (Deryugin) [inline src/compat/libc/stdio/rename_dvfs.c]: Yes, use `dirname()` here, please.

File: src/drivers/block_dev/block_dev.c [hunk 7]
```diff
@@ -54,7 +59,7 @@ static int block_dev_cache_free(void *dev) {
 	return  0;
 }
 
-struct block_dev *block_dev_create_common(const char *path, void *driver, void *privdata) {
+static struct block_dev *__block_dev_create(const char *path, const struct block_dev_driver *driver, void *privdata) {
```
Comment (Deryugin) [inline src/drivers/block_dev/block_dev.c]: Why not just put this code directly into `block_dev_create()`?

**Pull request merged**